code/modules/client/preferences/ai_core_display.dm

```diff
@@ -12,6 +12,7 @@
 
 	for (var/screen in GLOB.ai_core_display_screens - "Portrait" - "Random")
 		values[screen] = icon('icons/mob/silicon/ai.dmi', resolve_ai_icon_sync(screen))
+		CHECK_TICK
```

Comment:
do not CHECK_TICK in a single thing in preferences or even assets as a whole.

i have tried three times. every time byond shits itself and gives completely useless and unsolvable icon errors. theres a very good reason it does what it does

Reply:
this may have been reduced when I did the icon refreshing thing. Not sure how sleeping precisely here will play with that tho